base/io.jl

```diff
@@ -174,7 +174,12 @@ Read a single line of text, including a trailing newline character (if one is re
 the end of the input), from the given I/O stream or file (defaults to `STDIN`).
 When reading from a file, the text is assumed to be encoded in UTF-8.
 """
-readline(filename::AbstractString) = open(readline, filename)
+function readline(filename::AbstractString, chomp = false; nl2lf = false)
```

Comment:
I don't think we need to accept a `newlines` argument to `readline` if it can't be made efficient. People can use `readuntil` for special needs.